### PR TITLE
[Disco] Add LoadAll method to Disco Shard Loader

### PR DESCRIPTION
**Pull request by jinhongyii**
This PR adds a `LoadAll` method to the Disco shard loader to load
parameters all at once.

### CODE REVIEW

**Comment by jinhongyii** [comment]
cc: @junrushao @tqchen

File: src/runtime/disco/loader.cc [hunk 1]
```diff
@@ -93,9 +93,18 @@ inline std::vector<ShapeTuple::index_type> ShardShape(const ShapeTuple& shape, i
   return result;
 }
 
+class ShardInfoCompare {
+  public:
+    bool operator()(const ShardLoaderObj::ShardInfo& a, const ShardLoaderObj::ShardInfo& b) const {
+      int a_id = std::stoi(a.param->name.substr(6));
+      int b_id = std::stoi(b.param->name.substr(6));
+      return a_id < b_id;
```
Comment (junrushao) [inline src/runtime/disco/loader.cc:101]: you don't need an actual "sort", instead you may retrieve the parameter names in ndarray-cache-metadata and calculate their names correspondingly by "params_0", "params_1", etc

File: src/runtime/disco/loader.cc [hunk 2]
```diff
@@ -43,7 +43,7 @@ class ShardLoaderObj : public Object {
   /*! \brief Create a shard loader. */
   static ObjectRef Create(const std::string& path_to_metadata, const std::string& metadata,
                           const std::string& shard_info,
-                          TypedPackedFunc<void(DLTensor*, int, DLTensor*)> f_shard);
+                          PackedFunc f_shard);
```
Comment (junrushao) [inline src/runtime/disco/loader.cc:46]: Why this change?

Comment (jinhongyii) [reply src/runtime/disco/loader.cc:46]: TypedPackedFunc cannot be null

File: src/runtime/disco/loader.cc [hunk 3]
```diff
@@ -187,5 +231,16 @@ TVM_REGISTER_GLOBAL("runtime.disco.ShardLoaderLoad")
       return loader->Load(IntegerFromShapeTuple(weight_index));
     });
 
+TVM_REGISTER_GLOBAL("runtime.disco.ShardLoaderLoadAll").set_body_typed([](ObjectRef loader_obj) {
+  const auto* loader = loader_obj.as<ShardLoaderObj>();
+  CHECK(loader != nullptr) << "TypeError: Expected ShardLoaderObj, but gets: "
+                           << loader_obj->GetTypeKey();
+  Array<NDArray> shards;
+  for (int i = 0; i < static_cast<int>(loader->shard_info_.size()); i++) {
```
Comment (junrushao) [inline src/runtime/disco/loader.cc:239]: does `shard_info_.size()` imply all parameters?

**Comment by junrushao** [comment]
To facilitate PR merge, I made the related changes accordingly which will save @jinhongyii's time
